validate/validate.go

```diff
@@ -903,14 +903,14 @@ func namespaceValid(ns rspec.LinuxNamespace) bool {
 	case rspec.UserNamespace:
 	case rspec.CgroupNamespace:
 	default:
-		return false
+		return fmt.Errorf("invalid namespace type %s", ns.Type)
```

Comment:
The spec actually has no RFC2119 wording for this check, saying only “[The following namespace types are supported][1]”.  It doesn't say that runtimes MUST support them, or that they MUST NOT support additional values, or even that configurations SHOULD use values from that list.  I'd guess the spec-maintainer intention was closer to:

> Configurations SHOULD use, and runtimes SHOULD support, the following values:

Compliance testing would be easier with “runtimes MUST support”, but see opencontainers/runtime-spec#813.

Anyhow, from the config-validation side, I'd be happier seeing this supported by an effective-SHOULD, similar to what I've proposed in #481.

[1]: https://github.com/opencontainers/runtime-spec/blame/v1.0.0/config-linux.md#L29-L36